warehouse/postgres/postgres.go

```diff
@@ -0,0 +1,520 @@
+		logger.Errorf("PG: Error in downloading file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+		return "", err
+	}
+	fileName := objectFile.Name()
+	if err = objectFile.Close(); err != nil {
+		logger.Errorf("PG: Error in closing downloaded file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+		return "", err
+	}
+	return fileName, nil
+}
+
+func (pg *HandleT) loadTable(tableName string, columnMap map[string]string) (err error) {
+	status, err := warehouseutils.GetTableUploadStatus(pg.Upload.ID, tableName, pg.DbHandle)
+	if status == warehouseutils.ExportedDataState {
+		logger.Infof("PG: Skipping load for table:%s as it has been successfully loaded earlier", tableName)
+		return
+	}
+	if !warehouseutils.HasLoadFiles(pg.DbHandle, pg.Warehouse.Source.ID, pg.Warehouse.Destination.ID, tableName, pg.Upload.StartLoadFileID, pg.Upload.EndLoadFileID) {
+		warehouseutils.SetTableUploadStatus(warehouseutils.ExportedDataState, pg.Upload.ID, tableName, pg.DbHandle)
+		return
+	}
+	logger.Infof("PG: Starting load for table:%s", tableName)
+	warehouseutils.SetTableUploadStatus(warehouseutils.ExecutingState, pg.Upload.ID, tableName, pg.DbHandle)
+	timer := warehouseutils.DestStat(stats.TimerType, "single_table_upload_time", pg.Warehouse.Destination.ID)
+	timer.Start()
+	defer timer.End()
+
+	dbHandle, err := connect(pg.getConnectionCredentials(optionalCredsT{schemaName: pg.Namespace}))
+	if err != nil {
+		logger.Errorf("PG: Error establishing connection for copying table:%s: %v", tableName, err)
+		warehouseutils.SetTableUploadError(warehouseutils.ExportingDataFailedState, pg.Upload.ID, tableName, err, pg.DbHandle)
+		return
+	}
+	defer dbHandle.Close()
+	objectFileName, err := pg.DownloadLoadFile(tableName)
+	if err != nil {
+		warehouseutils.SetTableUploadError(warehouseutils.ExportingDataFailedState, pg.Upload.ID, tableName, err, pg.DbHandle)
+		return
+	}
+
+	gzipFile, err := os.Open(objectFileName)
+	if err != nil {
+		logger.Errorf("WH: Error opening file using os.Open for file:%s while loading to table %s", objectFileName, tableName)
+		warehouseutils.SetTableUploadError(warehouseutils.ExportingDataFailedState, pg.Upload.ID, tableName, err, pg.DbHandle)
+		return
+
+	}
+	defer gzipFile.Close()
+	gzipReader, err := gzip.NewReader(gzipFile)
+	if err != nil {
+		if err.Error() == "EOF" {
+			logger.Infof("WH: EOF while reading file using gzip.NewReader for file:%s while loading to table %s", gzipFile, tableName)
+		} else {
+			logger.Errorf("WH: Error reading file using gzip.NewReader for file:%s while loading to table %s", gzipFile, tableName)
+			warehouseutils.SetTableUploadError(warehouseutils.ExportingDataFailedState, pg.Upload.ID, tableName, err, pg.DbHandle)
+			return
+
+		}
+
+	}
+
+	// sort column names
+	sortedColumnKeys := warehouseutils.SortColumnKeysFromColumnMap(columnMap, pg.Warehouse.Destination.DestinationDefinition.Name)
```

Comment:
can you also make the change to use this func in other warehouse destinations also